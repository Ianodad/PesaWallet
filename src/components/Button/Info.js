import React from 'react';
import {StyleSheet, View} from 'react-native';
import {color} from 'react-native-reanimated';
import defaultStyles from '../../config/styles';
import Text from '../Text';

const Info = ({style, subTitle, subStyle, detail, detailStyle}) => {
  return (
    <View style={[styles.container, style]}>
      {subTitle && <Text style={[styles.subTitle, subStyle]}>{subTitle}</Text>}
      {detail && (
        <Text style={[styles.detailStyle, detailStyle]}>{detail}</Text>
      )}
    </View>
  );
};

export default Info;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
  },
  subTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: 'white',
  },
  detailStyle: {
    fontWeight: 'bold',
    color: 'white',
  },
});
