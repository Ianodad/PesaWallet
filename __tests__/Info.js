import React from 'react';
import Info from '../src/components/Button/Info';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';

import toJson from 'enzyme-to-json';

const defaultProps = {
  style: {margin: '10px'},
  subTitle: 'subTitle',
  subStyle: {margin: '10px'},
  detail: 'detail',
  detailStyle: {margin: '10px'},
};

describe('<Button/>', () => {
  const fn = jest.fn();
  const image = jest.mock('../src/assets/bootsplash_logo.png');
  const subTitle = 'subTitle';
  const detail = 'detail';
  describe('Rendering with snapshot', () => {
    // should render with any outcome
    it('should match to snapshot with crashing', () => {
      // eslint-disable-next-line no-undef
      const component = shallow(<Info />);
      expect(component).toMatchSnapshot();
    });

    it('should match subTitle snapshot with crashing', () => {
      // eslint-disable-next-line no-undef
      const component = shallow(<Info subTitle={subTitle} />);
      expect(component).toMatchSnapshot();
    });

    it('should match detail snapshot with crashing', () => {
      // eslint-disable-next-line no-undef
      const component = shallow(<Info detail={detail} />);
      expect(component).toMatchSnapshot();
    });

    it('should render with title snapshot', () => {
      const component = shallow(<Info {...defaultProps} />);
      expect(component).toMatchSnapshot();
    });

    describe('Rendering with unit testing', () => {
      it('find Element', () => {
        let component = renderer.create(<Info subTitle={'subTitle'} />);
        // get  the passed props from the component
        let instanceProps = component.root.findByType(Text).props.children;
        expect(instanceProps).toEqual('subTitle');
      });
    });
  });
});
