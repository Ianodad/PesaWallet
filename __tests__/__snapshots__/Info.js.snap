// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Rendering with snapshot should match detail snapshot with crashing 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should match subTitle snapshot with crashing 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should match to snapshot with crashing 1`] = `ShallowWrapper {}`;

exports[`<Button/> Rendering with snapshot should render with title snapshot 1`] = `ShallowWrapper {}`;
